sklearn/neighbors/_base.py

```diff
@@ -385,6 +386,17 @@ def _radius_neighbors_from_graph(graph, radius, return_distance):
 
 class NeighborsBase(MultiOutputMixin, BaseEstimator, metaclass=ABCMeta):
     """Base class for nearest neighbors estimators."""
+    _parameter_constraints = {
+        "n_neighbors": [Interval(Integral, 1, None, closed="left")],
+        "radius": [Interval(Integral, 1, None, closed="left"), None],
+        "weights": [StrOptions({"uniform", "distance"}), callable],
+        "algorithm": [StrOptions({"auto", "ball_tree", "kd_tree", "brute"}), callable],
```

Comment:
From my understanding, this replaces the checking in the first 2 lines of [_check_algorithm_metric](https://github.com/scikit-learn/scikit-learn/pull/23511/files#diff-945a1221f914501203afb06a3bc50d8bff40dcd6fd29571ad646267001e27697R424-R425):

```python
    if self.algorithm not in ["auto", "brute", "kd_tree", "ball_tree"]:
        raise ValueError("unrecognized algorithm: '%s'" % self.algorithm)
```